PGM/src/main/java/tc/oc/pgm/shop/SimplePurchaseTracker.java

```diff
@@ -0,0 +1,141 @@
+import tc.oc.pgm.shop.strategy.PartyPaymentStrategy;
+import tc.oc.pgm.shop.strategy.PaymentStrategy;
+import tc.oc.pgm.shop.strategy.PlayerPaymentStrategy;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Simple implementation of the {@link PurchaseTracker}.
+ */
+class SimplePurchaseTracker implements PurchaseTracker {
+    private final Map<MatchPlayer, PaymentStrategy> individualStrategies;
+    private final Map<Party, PaymentStrategy> partyStrategies;
+    private final Set<PaymentStrategy> globalStrategies;
+
+    private final boolean crossParty;
+    private final boolean persistent;
+
+    public SimplePurchaseTracker(boolean crossParty, boolean persistent) {
+        this.individualStrategies = Maps.newHashMap();
+        this.partyStrategies = Maps.newHashMap();
+        this.globalStrategies = Sets.newHashSet();
+        this.crossParty = crossParty;
+        this.persistent = persistent;
+    }
+
+    @Override
+    public void startIndividualStrategy(MatchPlayer player, Purchasable purchasable) {
+        startIndividualStrategy(player, purchasable, purchasable.getPurchaseFilter());
+    }
+
+    @Override
+    public void startPartyStrategy(Party party, MatchPlayer player, Purchasable purchasable) {
+        startPartyStrategy(party, player, purchasable, purchasable.getPurchaseFilter());
+    }
+
+    @Override
+    public void startGlobalStrategy(Purchasable purchasable, MatchPlayer player) {
+        startGlobalStrategy(purchasable, player, purchasable.getPurchaseFilter());
+    }
+
+    @Override
+    public void startIndividualStrategy(MatchPlayer player, Purchasable purchasable, Filter filter) {
+        PlayerPaymentStrategy strategy = new PlayerPaymentStrategy(purchasable, filter, player, crossParty, persistent);
+        individualStrategies.put(player, strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public void startPartyStrategy(Party party, MatchPlayer player, Purchasable purchasable, Filter filter) {
+        PartyPaymentStrategy strategy = new PartyPaymentStrategy(purchasable, filter, party);
+        partyStrategies.put(party, strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public void startGlobalStrategy(Purchasable purchasable, MatchPlayer player, Filter filter) {
+        GlobalPaymentStrategy strategy = new GlobalPaymentStrategy(purchasable, filter);
+        globalStrategies.add(strategy);
+        contribute(strategy, player);
+    }
+
+    @Override
+    public Optional<PaymentStrategy> getOngoingStrategy(MatchPlayer player, Purchasable purchasable) {
+        return individualStrategies.values().stream().filter(e ->
+            e.getOwner().equals(player) && e.getPurchasable().equals(purchasable)
+        ).findFirst();
+    }
+
+    @Override
+    public Optional<PaymentStrategy> getOngoingStrategy(Party party, Purchasable purchasable) {
```

Comment:
Wanted to leave this as abstract as possible for future stuff where parties could be used for different things. Also I believe somewhere I couldn't get a competitor from a player but could get a party so I used that.